subsys/shell/shell_dummy.c

```diff
@@ -55,13 +57,21 @@ static int write(const struct shell_transport *transport,
 		 const void *data, size_t length, size_t *cnt)
 {
 	struct shell_dummy *sh_dummy = (struct shell_dummy *)transport->ctx;
+	size_t store_cnt;
 
 	if (!sh_dummy->initialized) {
 		*cnt = 0;
 		return -ENODEV;
 	}
 
+	store_cnt = length;
+	if (sh_dummy->len + store_cnt >= sizeof(sh_dummy->buf))
```

Comment:
need brackets around if statement:
```
+       if (sh_dummy->len + store_cnt >= sizeof(sh_dummy->buf)) {
+               store_cnt = sizeof(sh_dummy->buf) - sh_dummy->len - 1;
+       }
```

Reply:
OK you mean braces, I see. I thought you wanted extra brackets in the if() expression and was struggling to figure that out.